cc/tf_bt_utils.cc

```diff
@@ -82,5 +91,92 @@ void WriteGameExamples(const std::string& gcp_project_name,
             << examples.size() << std::endl;
 }
 
+uint64_t IncrementGameCounter(const std::string& gcp_project_name,
+                              const std::string& instance_name,
+                              const std::string& table_name, size_t delta) {
+  Table table(CreateDefaultDataClient(gcp_project_name, instance_name,
+                                      google::cloud::bigtable::ClientOptions()),
+              table_name);
+  using namespace google::cloud::bigtable;
+  auto rule =
+      ReadModifyWriteRule::IncrementAmount("metadata", "game_counter", delta);
+  auto row = table.ReadModifyWriteRow("table_state", rule);
+  for (auto const& cell : row.cells()) {
+    if (cell.family_name() == "metadata" &&
+        cell.column_qualifier() == "game_counter") {
+      return cell.value_as<bigendian64_t>().get();
+    }
+  }
+  MG_FATAL() << "Failed to increment table_state=metadata:game_counter";
+  return 0;
+}
+
+void PortGamesToBigtable(const std::string& gcp_project_name,
+                         const std::string& instance_name,
+                         const std::string& table_name,
+                         const std::vector<std::string>& paths,
+                         int64_t game_counter) {
+  auto client_options = google::cloud::bigtable::ClientOptions();
+  auto channel_args = client_options.channel_arguments();
+  channel_args.SetUserAgentPrefix("minigo_to_cbt");
+  channel_args.SetInt(GRPC_ARG_ENABLE_CENSUS, 0);
+  channel_args.SetInt(GRPC_ARG_KEEPALIVE_PERMIT_WITHOUT_CALLS, 0);
+  channel_args.SetInt(GRPC_ARG_KEEPALIVE_TIMEOUT_MS, 60 * 1000);
+  client_options.set_channel_arguments(channel_args);
+  Table table(
+      CreateDefaultDataClient(gcp_project_name, instance_name, client_options),
+      table_name);
+
+  if (game_counter < 0) {
+    if (paths.size() != 1) {
+      MG_FATAL() << "Atomic game updates require batch size of 1 game";
+      return;
+    }
+    MG_FATAL() << "Have not yet implemented atomic game counter update.";
+    return;
+  }
+
+  google::cloud::bigtable::BulkMutation game_batch;
+  auto start_time = absl::Now();
+  int changes = 0;
+  for (const auto& path : paths) {
+    std::unique_ptr<tensorflow::RandomAccessFile> file;
+    TF_CHECK_OK(tensorflow::Env::Default()->NewRandomAccessFile(path, &file));
+
+    RecordReaderOptions options;
```

Comment:
Nit: I think it's slightly preferred to use

`RecordReaderOptions options = RecordReaderOptions::CreateRecordReaderOptions("ZLIB")`

Reply:
Thanks! Will fix.

Reply:
Actually, now that I'm there, I realize that you're recommending not only the factory method, but taking the compression as a string rather than as an enumeration. I recognize that such a pattern is valuable when wanting to plumb through command-line arguments, putting the user easily in charge of the configuration, but when the format is known, wouldn't the enumeration be better?

Reply:
To use "GZIP" you should really set more than just the compression type hence the more general need for the helper method.

This is non-production code, on a 20% project. Do what makes you happier. Either is fine be me.